plugins/module_utils/module_helper.py

```diff
@@ -296,17 +296,25 @@ def find_format(_param):
         param_list = params if params else self.module.params.keys()
 
         for param in param_list:
-            if param in self.module.argument_spec:
-                if param not in self.module.params:
-                    continue
-                fmt = find_format(param)
-                value = self.module.params[param]
+            if isinstance(param, dict):
+                if len(param.keys()) != 1:
+                    raise ModuleHelperException("run_command parameter as a dict must "
+                                                "contain only one key: {0}".format(param))
+                _param = list(param.keys())[0]
+                fmt = find_format(_param)
+                value = param[_param]
+            elif isinstance(param, str):
+                if param in self.module.argument_spec:
+                    fmt = find_format(param)
+                    value = self.module.params[param]
```

Comment:
The old code did the following:
```suggestion
                    if param not in self.module.params:
                        continue
                    value = self.module.params[param]
```

Reply:
Looking at it right now, I believe that *was* redundant in the previous code, because in:
```python
            if param in self.module.argument_spec:
                if param not in self.module.params:
```
if the parameter was in `argument_spec`, then it would *always* be in `module.params`, even if having `None` for value. Thatś the reason why we have to always check `is None` instead of `in module.params`. Right?

That said, the case when `module.params[param] is None` would run the assignments before, basically delegating the decision of what to do with `None` to the `ArgFormat` supporting that parameter. The new code behaves the same way for that case.

Reply:
Ah, great, that's good :)